rbd/mirror.go

```diff
@@ -57,6 +58,8 @@ const (
 	// ImageMirrorModeSnapshot uses snapshots to propagate RBD images between
 	// ceph clusters.
 	ImageMirrorModeSnapshot = ImageMirrorMode(C.RBD_MIRROR_IMAGE_MODE_SNAPSHOT)
+	// ImageMirrorModeAny is only used in requests to select any mode.
+	ImageMirrorModeAny = ImageMirrorMode(math.MinInt64)
```

Comment:
What's the rationale behind MinInt64? Is it just to avoid colliding with some value ceph might use?

Maybe this is a lame idea but perhaps instead of trying to define a "fake" ImageMirrorMode for this we could do something like:
```
type MirrorImageInfoFilter type {
    Enable bool
    Mode ImageMirrorMode
}
```

And pass a value with tihs type as an argument? Would it avoid the weirdness with creating a pointer to int so that none is "missing/unset"? If not, I've no big issues with this just figured it would be food for thought.

Reply:
In general I'm all in favour for an "optional type" as you might remember, but this seems very verbose for a simple integer parameter. In other cases we chose the "zero value" of a parameter (usually "" string) as the "none" value, but here unfortunately it's taken already. I can also take -1, I really don't care. ~The "right" thing for me would be to use a pointer, because that is arguably the Go idiomatic optional type. (Actually, I noticed that you _did_ use a pointer as an option as well as return value of `Next()`, what's going on? 😜 )~ While I was writing this I had another idea, let me try...

Reply:
The pointer return from Next() was to be consistent with the way we implemented the Next() function elsewhere in go-ceph, so in short - to be consistent. It was not my first choice, but as we've discussed without a common standard, esp. one that works with something like range, for iteration - I preferred to be "self consistent" with previous work in go-ceph. :-)

Reply:
Implemented the option as interface, that can be `nil`